tianshou/data/utils/converter.py

```diff
@@ -80,3 +82,64 @@ def to_torch_as(
     """
     assert isinstance(y, torch.Tensor)
     return to_torch(x, dtype=y.dtype, device=y.device)
+
+
+# Note: object is used as a proxy for objects that can be pickled
+Hdf5ConvertibleValues = Union[
+    int, float, Batch, np.ndarray, torch.Tensor, object, 'Hdf5ConvertibleType'
+]
+Hdf5ConvertibleType = Dict[str, Hdf5ConvertibleValues]
+
+
+def to_hdf5(x: Hdf5ConvertibleType, y: h5py.Group) -> None:
+    """Copy object into HDF5 group."""
+    for k, v in x.items():
+        # dicts and batches are both represented by groups
+        if isinstance(v, (Batch, dict)):
+            subgrp = y.create_group(k)
+            if isinstance(v, dict):
+                subgrp_data = v
+            else:
+                subgrp_data = v.__getstate__()
+                subgrp.attrs["__data_type__"] = "Batch"
+            to_hdf5(subgrp_data, subgrp)
+        # numpy arrays and pytorch tensors are written to datasets
+        elif isinstance(v, np.ndarray):
+            y.create_dataset(k, data=v)
+            y[k].attrs["__data_type__"] = "numpy"
+        elif isinstance(v, torch.Tensor):
+            y.create_dataset(k, data=v.cpu().numpy())
+            y[k].attrs["__data_type__"] = "torch"
+        # ints and floats are stored as attributes of groups
+        elif isinstance(v, (int, float)):
+            y.attrs[k] = v
+        # resort to pickle for any other type of object
+        else:
+            int_data = np.fromstring(pickle.dumps(v), dtype="uint8")
```

Comment:
I guess this could fail ? If so, wrap it in a `try ... except ...` block, and raise `NotImplementedError`, describing that the object is not compatible with hdf5 import/export format. Try to make as clear as possible for the user to understand what he did wrong.

Reply:
Yes, makes sense. I have added a try-catch block with a `NotImplementedError`. Let me know if the message makes sense to you.